File: ki/paiza4.py
N = int(input())
adjacent_list = [ [] for i in range(N)]
# print(adjacent_list)
for i in range(N - 1):
  a, b = map(int, input().split())
  adjacent_list[a - 1].append(b)
  adjacent_list[b - 1].append(a)

for i in range(N):
  if len(adjacent_list[i]) == 1:
    print(i + 1)

File: sennkeitansaku2/paiza4.swift
import Foundation

func filterRowsByColumnValue(n: Int, m: Int, columnIndex: Int, threshold: Int) -> [String] {
    var filteredRows: [String] = []
    
    for _ in 0..<n {
        let row = readLine()!.split(separator: ",").map { Int($0)! }
        if row[columnIndex - 1] >= threshold {
            let rowWithComma = row.map { String($0) }.joined(separator: ",")
            filteredRows.append(rowWithComma)
        }
    }
    return filteredRows
}

let input = readLine()!.split(separator: " ").map { Int($0)! }
let (n, m, columnIndex, threshold) = (input[0], input[1], input[2], input[3])

let result = filterRowsByColumnValue(n: n, m: m, columnIndex: columnIndex, threshold: threshold)
result.forEach {
    print($0)
}